extensions/pkg/controller/backupentry/genericactuator/actuator.go

```diff
@@ -101,8 +105,17 @@ func (a *actuator) Delete(ctx context.Context, log logr.Logger, be *extensionsv1
 	return a.backupEntryDelegate.Delete(ctx, log, be)
 }
 
-func (a *actuator) deleteEtcdBackupSecret(ctx context.Context, secretName string) error {
-	etcdSecret := emptyEtcdBackupSecret(secretName)
+func (a *actuator) deleteEtcdBackupSecret(ctx context.Context, backupEntryName string) error {
+	etcdSecret := emptyEtcdBackupSecret(backupEntryName)
+	if err := a.client.Get(ctx, client.ObjectKeyFromObject(etcdSecret), etcdSecret); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("could not delete secret %s in %s namespace: %w", etcdSecret.Name, etcdSecret.Namespace, err)
+	}
+	if createdBy, ok := etcdSecret.Annotations[AnnotationKeyCreatedByBackupEntry]; ok && createdBy != backupEntryName {
+		return nil
+	}
 	return kubernetesutils.DeleteObject(ctx, a.client, etcdSecret)
```

Comment:
You could add empty lines to make it more readable.

```suggestion

	if createdBy, ok := etcdSecret.Annotations[AnnotationKeyCreatedByBackupEntry]; ok && createdBy != backupEntryName {
		return nil
	}

	return kubernetesutils.DeleteObject(ctx, a.client, etcdSecret)
```